godot-macros/src/godot_api.rs

```diff
@@ -46,23 +48,25 @@ pub fn transform(input_decl: Declaration) -> Result<TokenStream, Error> {
 
 /// Attribute for user-declared function
 enum BoundAttrType {
-    Func(AttributeValue),
-    Signal(AttributeValue),
-    Const(AttributeValue),
+    Func { rename: Option<String> },
+    Signal,
+    Const,
 }
 
-struct BoundAttr {
-    attr_name: Ident,
+struct BoundAttr<'a> {
+    attr_name: &'a str,
     index: usize,
     ty: BoundAttrType,
 }
 
-impl BoundAttr {
+impl BoundAttr<'_> {
     fn bail<R>(self, msg: &str, method: &Function) -> Result<R, Error> {
         bail!(&method.name, "#[{}]: {}", self.attr_name, msg)
     }
 }
 
+type Signal = Function;
+
```

Comment:
The [original conversation history](https://github.com/godot-rust/gdext/pull/376#discussion_r1288089367) is in the other PR, but my comment there was:

> I'd probably keep this as a separate type, as the possible options for #[func] and #[signal] likely diverge over time. It also slightly enhances type safety.

and your response



> Is it not worth it to rename the type for a bit more clarity in the method signature for?
>
> fn process_godot_fns(decl: &mut Impl) -> Result<(Vec<FuncDefinition>, Vec<Signal>), Error>
> vs
> fn process_godot_fns(decl: &mut Impl) -> Result<(Vec<FuncDefinition>, Vec<Function>), Error>

Using `Signal` is more confusing because we're referring to venial symbols here, and they don't have that concept. So we mix a raw venial symbol `Function`  with our own semantic interpretation `Signal` in the same type.

Instead, the tuple `(Vec<FuncDefinition>, Vec<Signal>)` could be its own struct with named fields. Then we know from the field name what it means and don't need aliases (which have no type safety anyway).